### PR TITLE
Tests: add tests for shadow DOM roots

### PR DESCRIPTION
**Pull request by timmywil**
- This adds some basic tests to ensure we don't break on shadow roots.

Fixes gh-296

### CODE REVIEW

File: test/unit/selector.js [hunk 1]
```diff
@@ -1326,3 +1326,19 @@ test("select() with pre-compiled function", function() {
 		equal( Sizzle.select( compiled, Sizzle( "#first")[0] ).length, 0, "Should not match with different context" );
 	});
 });
+
+test("Sizzle supports shadow DOM nodes as root", function() {
+	if ( !document.documentElement.createShadowRoot ) {
+		ok( true, "Shadow DOM not supported in this browser" );
+		return;
```
Comment (timmywil) [inline test/unit/selector.js]: We should probably start using the same QUnit syntax in Sizzle at some point. We'll use skip here when that happens.

Comment (mzgol) [inline test/unit/selector.js]: We should also stop using the globals as we've done in jQuery. `QUnit.test` instead of `test`, `assert` as a test function parameter etc. Perhaps @markelog could apply one of those magic scripts that he used for jQuery. :-)

Comment (timmywil) [inline test/unit/selector.js]: Yea, that's what I mean, but that can be a separate PR.

Comment (mzgol) [inline test/unit/selector.js]: Yeah, definitely.

-- 
Michał Gołębiowski

Comment (markelog) [inline test/unit/selector.js]: Yeah, would you mind creating an issue about this, so i wouldn't forget?

Comment (mgol) [inline test/unit/selector.js]: Sorry for the delay, only now have I returned to this open tab. :-) #376
